services/migration_service/data/postgres_async_db.py

```diff
@@ -35,15 +35,18 @@ def __init__(self):
         AsyncPostgresDB.__instance = self
 
     async def _init(self, db_conf: DBConfiguration):
-        # todo make poolsize min and max configurable as well as timeout
-        # todo add retry and better error message
-        retries = 3
-        for i in range(retries):
+        # todo make poolsize min and max configurable
+        max_attempts = 3
+        while max_attempts > 0:
             try:
                 self.pool = await aiopg.create_pool(db_conf.get_dsn(), timeout=db_conf.timeout)
+                break
             except Exception as e:
                 print("printing connection exception: " + str(e))
-                if retries - i < 1:
```

Comment:
I see this is never actually hit as range excludes the end, but can be fixed with a slight tweak f.ex. 

```
if retries-(i+1) < 1:
```